macosx/Controller.m

```diff
@@ -3692,16 +3733,93 @@ - (NSRect) previewPanel: (QLPreviewPanel *) panel sourceFrameOnScreenForPreviewI
     }
 }
 
+- (IBAction) performTouchBarAction: (id) sender //senders here are always NSButtons, not NSTouchBarItems
+{
+    if (sender == createTouchBarButton)
+    {
+        [self createFile:sender];
+    }
+    if (sender == openFileTouchBarButton)
+    {
+        [self openShowSheet:sender];
+    }
+    if (sender == removeTouchBarButton)
+    {
+        [self removeNoDelete:sender];
+    }
+    if (sender == pauseAllTouchBarButton)
+    {
+        [self stopAllTorrents:sender];
+    }
+    if (sender == resumeAllTouchBarButton)
+    {
+        [self resumeAllTorrents:sender];
+    }
+    if (sender == speedLimitTouchBarButton)
+    {
+        [self toggleSpeedLimit: sender];
+    }
+    if (sender == cleanupTouchBarButton)
+    {
+        [self clearCompleted: sender];
+    }
+    if (sender == quickLookTouchBarButton)
+    {
+        [self toggleQuickLook:sender];
+    }
+    if (sender == showInfoTouchBarButton)
+    {
+        [self showInfo:sender];
+    }
+}
```

Comment:
You'd think that a switch would be best here. But unfortunately Objective-C doesn't let you match objects in a switch statement, only native C types like ints. It's an unfortunate quirk of the language.
As for the dictionary idea, there is a way to do what you suggest but it looks a bit messy in Objective C. Basically it involves storing the methods as Strings, converting them into Selectors and dispatching them at runtime. It's a bit fragile, but I admit it could scale better than the current solution… I'll look into it.